ReactiveCocoa/Swift/Atomic.swift

```diff
@@ -8,9 +8,56 @@
 
 import Foundation
 
+public protocol MutexType {
+	func lock()
+	func unlock()
+}
+
+public final class RecursiveLock: MutexType {
```

Comment:
I see your point guys, but I would prefer to refrain from changing particular recursive mutex implementation in this particular PR.